x-pack/plugins/security_solution/public/resolver/view/clickthrough.test.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { oneAncestorTwoChildren } from '../data_access_layer/mocks/one_ancestor_two_children';
+import { Simulator } from '../test_utilities/simulator';
+// Extend jest with a custom matcher
+import '../test_utilities/extend_jest';
+
+describe('Resolver, when analyzing a tree that has 1 ancestor and 2 children', () => {
+  let simulator: Simulator;
+  let databaseDocumentID: string;
+  let entityIDs: { origin: string; firstChild: string; secondChild: string };
+
+  // the resolver component instance ID, used by the react code to distinguish piece of global state from those used by other resolver instances
+  const resolverComponentInstanceID = 'resolverComponentInstanceID';
+
+  beforeEach(async () => {
+    // create a mock data access layer
+    const { metadata: dataAccessLayerMetadata, dataAccessLayer } = oneAncestorTwoChildren();
+
+    // save a reference to the entity IDs exposed by the mock data layer
+    entityIDs = dataAccessLayerMetadata.entityIDs;
+
+    // save a reference to the `_id` supported by the mock data layer
+    databaseDocumentID = dataAccessLayerMetadata.databaseDocumentID;
+
+    // create a resolver simulator, using the data access layer and an arbitrary component instance ID
+    simulator = new Simulator({ databaseDocumentID, dataAccessLayer, resolverComponentInstanceID });
+  });
+
+  describe('when it has loaded', () => {
+    beforeEach(async () => {
+      await expect(
+        /**
+         * It's important that all of these are done in a single `expect`.
+         * If you do them concurrently with each other, you'll have incorrect results.
+         *
+         * For example, there might be no loading element at one point, and 1 graph element at one point, but never a single time when there is both 1 graph element and 0 loading elements.
+         */
+        simulator.mapStateTransisions(() => ({
+          graphElements: simulator.graphElement().length,
+          graphLoadingElements: simulator.graphLoadingElement().length,
+          graphErrorElements: simulator.graphErrorElement().length,
+        }))
+      ).toSometimesYieldEqualTo({
```

Comment:
:grey_question: what? I think I get the part about transitioning to this, but the word `sometimes` is making it difficult... do we mean something like `finally` or even `at least once`?

Reply:
i need better names. was hoping to come up w/ then during an in-person code review.

Reply:
so here's my logic:
`Equal` has significance in `jest` matchers. I'm trying to express that this is using a 'deep equal' type comparison.

`Sometimes` is meant to evoke the array method [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) 

Here's the basic behavior:

```
    // Set to true if the test passes.
    let pass: boolean = false;

    // Async iterate over the iterable
    for await (const received of receivedIterable) {
      // keep track of the last value. Used in both pass and fail messages
      lastReceived = received;
      // Use deep equals to compare the value to the expected value
      if (this.equals(received, expected)) {
        // If the value is equal, break
        pass = true;
        break;
      }
    }
```

Reply:
`toYieldAtLeastOnce` sounds good. but it would be nice if the name indicated that deep equal was being used.

Reply:
OK, I think I understand better now. For testing purposes can we put a tighter collar on it; Like we care that it yielded this `at least once` but I think we care even more that it yielded equal  _last_, right? So maybe `finallyYieldedEqualTo` which captures the timing component, too.

Reply:
Or `eventuallyYieldedEqualTo`

Reply:
when with `toYieldEqualTo` for now. Let's continue improving this after merge